config/webpack.config.api.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+// Do this as the first thing so that any code reading it knows the right env.
+process.env.NODE_ENV = process.env.NODE_ENV || "development";
+process.env.BABEL_ENV = process.env.NODE_ENV;
+
+const webpack = require("webpack");
+const ModuleScopePlugin = require("react-dev-utils/ModuleScopePlugin");
+const RemoveBuildDirectoryPlugin = require("./RemoveBuildDirectoryPlugin");
+const paths = require("./paths");
+const getClientEnvironment = require("./env");
+
+const env = getClientEnvironment(null);
+
+module.exports = {
+  // Don't attempt to continue if there are any errors.
+  bail: true,
+  node: {
+    // Don't munge `__dirname` and `__filename`.
+    // https://github.com/webpack/webpack/issues/1599#issuecomment-186841345
+    __dirname: false,
+    __filename: false,
+  },
+  entry: {
+    api: paths.apiIndexJs,
+  },
+  output: {
+    path: paths.apiBuild,
+    // Generated JS file names (with nested folders).
+    // There will be one main bundle, and one file per asynchronous chunk.
+    // We don't currently advertise code splitting but Webpack supports it.
+    filename: "[name].js",
+    chunkFilename: "[name].[chunkhash:8].chunk.js",
+    libraryTarget: "umd",
```

Comment:
Using the `umd` library target, when not using a require function like Observables does but including it through a `<script>` tag, this will set a global variable. Currently this global is `window.default`. Because `index.js` exports `default`.

We can define `library: "sourcecred",` to namespace this. Though we will still have the default in between.
For example `window.sourcecred.default.core.graph.Graph`.

A way around that would be export several const in `index.js`. Which would also address my comment in #1526

```js

export const analysis = deepFreeze({
  timeline: {
    timelineCred,
  },
});

export const core = deepFreeze({
  address,
  attribution: {
    markovChain,
    graphToMarkovChain,
  },
  graph,
});

export const plugins = deepFreeze({
  github: {
    declaration: githubDeclaration,
    edges: githubEdges,
    nodes: githubNodes,
  },
  discourse: {
    address: discourseAddress,
    declaration: discourseDeclaration,
  },
});
```

Combined with `library: "sourcecred",` and a tweak to the observable example script that should be nice and consistent :]

Reply:
Noted. This doesn’t bother me because it’s not actually intended to be
used directly in a browser, so setting `default` isn’t a problem, and
splitting this out into separate exports would heterogenize the shape in
a way that really doesn’t align with the semantics and also introduces
extra work and typing, both here and in the Observable script. I think
that it’s fine as is.

Reply:
In that case, only setting `library: "sourcecred",` doesn't change the current behavior for Observables, but prevents loading as `window.default`.

Reply:
Okay, sure: added `library: "sourcecred"`.